src/modules/Elsa.Alterations/Middleware/Workflows/RunAlterationsMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+using Elsa.Alterations.Core.Contexts;
+using Elsa.Alterations.Core.Contracts;
+using Elsa.Alterations.Core.Models;
+using Elsa.Extensions;
+using Elsa.Workflows;
+using Elsa.Workflows.Pipelines.WorkflowExecution;
+
+namespace Elsa.Alterations.Middleware.Workflows;
+
+/// <summary>
+/// Middleware that runs alterations.
+/// </summary>
+internal class RunAlterationsMiddleware(WorkflowMiddlewareDelegate next, IEnumerable<IAlterationHandler> handlers) : WorkflowExecutionMiddleware(next)
+{
+    public static readonly object AlterationsPropertyKey = new();
+    public static readonly object AlterationsLogPropertyKey = new();
+
+    public override async ValueTask InvokeAsync(WorkflowExecutionContext context)
+    {
+        var alterations = (IEnumerable<IAlteration>)(context.TransientProperties.GetValue(AlterationsPropertyKey) ?? throw new InvalidOperationException("No alterations found in the transient properties."));
+        var log = (AlterationLog)(context.TransientProperties.GetValue(AlterationsLogPropertyKey) ?? throw new InvalidOperationException("No alteration log found in the transient properties."));
+        await RunAsync(context, alterations, log, context.CancellationTokens.ApplicationCancellationToken);
+    }
+
+    private async Task RunAsync(WorkflowExecutionContext workflowExecutionContext, IEnumerable<IAlteration> alterations, AlterationLog log, CancellationToken cancellationToken = default)
+    {
+        var commitActions = new List<Func<Task>>();
+
+        foreach (var alteration in alterations)
+        {
+            // Find handlers.
+            var supportedHandlers = handlers.Where(x => x.CanHandle(alteration)).ToList();
```

Comment:
Do we want to support having multiple handlers for the same type of alteration? I can imagine sequencing could have a major influence if we do.

Reply:
That's indeed the idea, although I haven't thought of a real use case for it yet. Can you think of a scenario where the order of handlers would have an impact? Because then we might also can think of a real use case, but handle it in a way where one can control the order of things. If not, then we could consider just supporting one handler per type.

Reply:
Given that this is not a functional change - just moved code, I will resolve this comment, but please do continue the conversation.